include/zephyr/bluetooth/mesh/access.h

```diff
@@ -673,6 +672,23 @@ struct bt_mesh_mod_id_vnd {
 };
 
 /** Abstraction that describes a Mesh Model instance */
+struct bt_mesh_model_data {
```

Comment:
The name slightly confusing with `bt_mesh_model_data_store` function. In the docs as model data we call a data, like states, that model operates on. Such data can also be stored in flash. So I would suggest to use some other name, e.g. `bt_mesh_model_ctx`, `bt_mesh_model_rt`, or something else.